karapace/protobuf/message_element.py

```diff
@@ -84,13 +84,14 @@ def to_schema(self) -> str:
         result.append("}\n")
         return "".join(result)
 
-    def compare(self, other: MessageElement, result: CompareResult, types: CompareTypes) -> None:
+    def compare(self, other: TypeElement, result: CompareResult, types: CompareTypes) -> None:
         from karapace.protobuf.compare_type_lists import compare_type_lists
 
+        if not isinstance(other, MessageElement):
+            result.add_modification(Modification.TYPE_ALTER)
```

Comment:
question, which is the semantics of the compare? 

In case of multiple changes, like for e.g. changing type and adding a field to the message do we just return the `TYPE_ALTER`?

Reply:
This line of code should be mostly unreachable as this is invoked from `compare_type_lists` and there is basically already `if` statement which checks both `self` and `other` of `compare` call are same type. But for safety it is better to repeat that here instead of adding `assert` IMHO because this would be right thing to do.